Build failed on Mac (x86_64-darwin) due to `ntohll(x)` and `htonll(x)` being `#defined` in `Libsystem`

Partial build log on OfBorg (the CI of [NixOS/nixpkgs](https://github.com/NixOS/nixpkgs)) that demonstrate the error:

```
/nix/store/b0q4wjdw6xwhfsqy8ab106cmhn5xrr8p-bash-5.1-p12/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I.  -I/nix/store/109pyfb5h9dally7iav62zcsg38i8ydw-dcap-2.47.12-dev/include -DLOCALEDIR=\"/nix/store/iyg74j1agq4iw0pvb1avk93hpfnvzdy4-dcap-2.47.12/share/locale\"   -D_REENTRANT -DLIBC_SYSCALLS -D_GNU_SOURCE -g -O2 -c -o dcap.lo dcap.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I/nix/store/109pyfb5h9dally7iav62zcsg38i8ydw-dcap-2.47.12-dev/include -DLOCALEDIR=\"/nix/store/iyg74j1agq4iw0pvb1avk93hpfnvzdy4-dcap-2.47.12/share/locale\" -D_REENTRANT -DLIBC_SYSCALLS -D_GNU_SOURCE -g -O2 -c dcap.c  -fno-common -DPIC -o .libs/dcap.o
In file included from dcap.c:56:
./dcap_functions.h:43:10: error: expected ')'
uint64_t ntohll(uint64_t x);
         ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:30: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
                             ^
./dcap_functions.h:43:10: note: to match this '('
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:5: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
    ^
In file included from dcap.c:56:
./dcap_functions.h:43:10: error: cannot redeclare builtin function '__builtin_constant_p'
uint64_t ntohll(uint64_t x);
         ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:6: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
     ^
./dcap_functions.h:43:10: note: '__builtin_constant_p' is a builtin with type 'int ()'
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:6: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
     ^
In file included from dcap.c:56:
./dcap_functions.h:47:10: error: expected ')'
uint64_t htonll(uint64_t arg);
         ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:141:25: note: expanded from macro 'htonll'
#define htonll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:30: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
                             ^
./dcap_functions.h:47:10: note: to match this '('
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:141:25: note: expanded from macro 'htonll'
#define htonll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:5: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
    ^
In file included from dcap.c:56:
./dcap_functions.h:47:10: error: cannot redeclare builtin function '__builtin_constant_p'
uint64_t htonll(uint64_t arg);
         ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:141:25: note: expanded from macro 'htonll'
#define htonll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:6: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
     ^
./dcap_functions.h:43:10: note: '__builtin_constant_p' is a builtin with type 'int ()'
uint64_t ntohll(uint64_t x);
         ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:6: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
     ^
dcap.c:130:15: error: expected ')'
    uint64_t  htonll(uint64_t);
              ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:141:25: note: expanded from macro 'htonll'
#define htonll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:30: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
                             ^
dcap.c:130:15: note: to match this '('
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:141:25: note: expanded from macro 'htonll'
#define htonll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:5: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
    ^
dcap.c:130:15: error: cannot redeclare builtin function '__builtin_constant_p'
    uint64_t  htonll(uint64_t);
              ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:141:25: note: expanded from macro 'htonll'
#define htonll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:6: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
     ^
./dcap_functions.h:43:10: note: '__builtin_constant_p' is a builtin with type 'int ()'
uint64_t ntohll(uint64_t x);
         ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:6: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
     ^
dcap.c:133:15: error: expected ')'
    uint64_t  ntohll(uint64_t);
              ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:30: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
                             ^
dcap.c:133:15: note: to match this '('
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:5: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
    ^
dcap.c:133:15: error: cannot redeclare builtin function '__builtin_constant_p'
    uint64_t  ntohll(uint64_t);
              ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:6: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
     ^
./dcap_functions.h:43:10: note: '__builtin_constant_p' is a builtin with type 'int ()'
uint64_t ntohll(uint64_t x);
         ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/nix/store/8xkv6d53n2hznyvf1vsyknqz4awj77i7-Libsystem-1238.60.2/include/libkern/_OSByteOrder.h:78:6: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
     ^
```

https://logs.nix.ci/?key=nixos/nixpkgs.156282&attempt_id=62c1024b-7e66-4cd7-ae7c-a69ad6d84d6b

Build expression: https://github.com/NixOS/nixpkgs/pull/156282

--- a/src/dcap.c
+++ b/src/dcap.c
@@ -127,10 +127,11 @@ static ConfirmationBlock get_reply(int);
 static int name_invalid(char *);
 static server * url2srv(dcap_url *);
 
-#ifndef HAVE_HTONLL
+#if !defined(HAVE_HTONLL) && !defined(htonll)
     uint64_t  htonll(uint64_t);
 #endif
-#ifndef HAVE_NTOHLL
+
+#if !defined(HAVE_NTOHLL) && !defined(ntohll)
     uint64_t  ntohll(uint64_t);
 #endif
 
@@ -1631,7 +1632,7 @@ get_fin(struct vsp_node * node)
 #  error Unknown Byte order
 #endif
 
-#ifndef HAVE_NTOHLL
+#if !defined(HAVE_NTOHLL) && !defined(ntohll)
     uint64_t
     ntohll(uint64_t x)
     {
@@ -1641,15 +1642,15 @@ get_fin(struct vsp_node * node)
         return x;
     #endif /* I_AM_LITTLE_ENDIAN */
     }
-#endif /* HAVE_NTOHLL */
+#endif /* HAVE_NTOHLL ntohll */
 
-#ifndef HAVE_HTONLL
+#if !defined(HAVE_HTONLL) && !defined(htonll) 
     uint64_t
     htonll(uint64_t arg)
     {
         return ntohll(arg);
     }
-#endif /* HAVE_HTONLL */
+#endif /* HAVE_HTONLL htonll */
 
 void dc_setReplyHostName(const char *s)
 {

--- a/src/dcap.c
+++ b/src/dcap.c
@@ -127,14 +127,6 @@ static ConfirmationBlock get_reply(int);
 static int name_invalid(char *);
 static server * url2srv(dcap_url *);
 
-#if !defined(HAVE_HTONLL) && !defined(htonll)
-    uint64_t  htonll(uint64_t);
-#endif
-
-#if !defined(HAVE_NTOHLL) && !defined(ntohll)
-    uint64_t  ntohll(uint64_t);
-#endif
-
 static MUTEX(couterLock);
 static unsigned int connectionCounter = 0;
 static unsigned int newCounter();

--- a/src/dcap_functions.h
+++ b/src/dcap_functions.h
@@ -39,10 +39,14 @@ int get_fin( struct vsp_node *);
 int get_ack(int , ConfirmationBlock * );
 
 
-#ifndef HAVE_NTOHLL
+// Declare the function if not defined by the system
+// as eithter a function or a macro
+// Darwin Libsystem may #define ntohll as a parameterized macro
+#if !defined(HAVE_NTOHLL) && !defined(ntohll)
 uint64_t ntohll(uint64_t x);
-#endif /* HAVE_NTOHLL */
+#endif /* HAVE_NTOHLL ntohll */
 
-#ifndef HAVE_HTONLL
+// Same as above`
+#if !defined(HAVE_HTONLL) && !defined(htonll)
 uint64_t htonll(uint64_t arg);
-#endif /* HAVE_HTONLL */
+#endif /* HAVE_HTONLL htonll */

